unified_planning/model/action.py

```diff
@@ -764,3 +764,228 @@ def __repr__(self) -> str:
         s.append("    ]\n")
         s.append("  }")
         return "".join(s)
+
+
+"""
+Below we have natural transitions. These are not controlled by the agent and would probably need a proper subclass. Natural transitions can be of two kinds:
+Processes or Events.
+Processes dictate how numeric variables evolve over time through the use of time-derivative functions
+Events dictate the analogous of urgent transitions in timed automata theory
+"""
+
+
+class Process(Action):
+    """This is the `Process` class, which implements the abstract `Process` class."""
```

Comment:
? It seems to me it implements the abstract `Action` class :smile:

I thinkit is not too bad to have processes and events to be `Action`s, even though they are semantically different. In hindsight we might have called the abstract action class somethig else.
@alvalentini maybe we can consider a refactoring of this name, eventually...

Reply:
One could call this transition, and I have hinted at something like that to Alessandro...but it is a major change no?